sympy/ntheory/factor_.py

```diff
@@ -1491,14 +1491,18 @@ class totient(Function):
     @classmethod
     def eval(cls, n):
         n = sympify(n)
-        if n.is_Integer:
+        if n==S.NegativeInfinity: raise ValueError("n must be a positive integer")
+        elif n==S.Infinity: return S.Infinity
+        elif n.is_Integer:
```

Comment:
This could be done more cleanly.  You can use `if n >= 1` initially and then

    try:
        factors = factorint(n)
        ...
        return t
     except ValueError:
        pass
     raise ValueError("n must be a positive integer, not {}".format(n))

That way you rely on `factorint`s integer checking and the error message is only repeated once.